step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java

```diff
@@ -172,47 +230,51 @@ private void persistStagedEntities(List<ExecutiontTaskParameters> completeExecTa
         }
 
         for (ExecutiontTaskParameters execTasksParameter : completeExecTasksParameters) {
-            executionScheduler.addExecutionTask(execTasksParameter, false);
+            if (executionScheduler != null) {
+                // TODO: move this to a dedicated class as part of SED-2594
+                executionScheduler.addExecutionTask(execTasksParameter, false);
+            } else {
+                executionTaskAccessor.save(execTasksParameter);
+            }
         }
 
     }
 
-    protected <T extends AbstractOrganizableObject & EnricheableObject> void fillEntities(List<T> entities, AutomationPackage newPackage, List<T> oldEntities, ObjectEnricher enricher){
-        Map<String, ObjectId> planNameToIdMap = createNameToIdMap(oldEntities);
+    protected <T extends AbstractOrganizableObject & EnricheableObject> void fillEntities(List<T> entities, List<T> oldEntities, ObjectEnricher enricher) {
+        Map<String, ObjectId> nameToIdMap = createNameToIdMap(oldEntities);
 
         for (T e : entities) {
             // keep old id
-            ObjectId oldId = planNameToIdMap.get(e.getAttribute(AbstractOrganizableObject.NAME));
+            ObjectId oldId = nameToIdMap.get(e.getAttribute(AbstractOrganizableObject.NAME));
             if (oldId != null) {
                 e.setId(oldId);
             }
 
-            e.addCustomField(AutomationPackageEntity.AUTOMATION_PACKAGE_ID, newPackage.getId().toString());
             if (enricher != null) {
                 enricher.accept(e);
             }
         }
     }
 
-    protected List<Plan> preparePlansStaging(AutomationPackage newPackage, AutomationPackageContent packageContent, AutomationPackage oldPackage, ObjectEnricher enricher) {
+    protected List<Plan> preparePlansStaging(AutomationPackageContent packageContent, AutomationPackage oldPackage, ObjectEnricher enricher) {
         List<Plan> plans = packageContent.getPlans();
-        fillEntities(plans, newPackage, oldPackage != null ? getPackagePlans(oldPackage) : new ArrayList<>(), enricher);
+        fillEntities(plans, oldPackage != null ? getPackagePlans(oldPackage.getId()) : new ArrayList<>(), enricher);
```

Comment:
@jeromecomte 
It seems we are not modifying old stored objects here. 

`getPackagePlans()` loads old plans via accessor (by the way, the local accessor should clone the original object because of `byPassCloning` property in `InMemoryCollection`). But these objects are not updated in `fillEntities()`. Those are only used to prepare the `nameToIdMap` and set the original ids to `plans` taken from `packageContent`,